torch/autograd/gradcheck.py

```diff
@@ -338,10 +380,12 @@ def fn(input):
         def fn(input):
             return _as_tuple(func(*input))[i]
 
+        numerical = get_numerical_jacobian(fn, tupled_inputs, eps=eps)
+
+        o = fn(tupled_inputs)
```

Comment:
Yes, The forward is re-done here to prevent failures for some sparse tests when the numerical jacobian is done before the backward analytical one.
I am not 100% sure what is the root cause. But re-doing the forward seems like the right thing to do in any case.